content/tutorials/Straightforward-Video-calls-with-WebRTC-and-Ably.textile

```diff
@@ -0,0 +1,408 @@
+
+We need to call the @AblyRealtime.presence.enter()@ method so Ably is aware that we want to enter into the presence channels and keep track of the new client.
+
+Lastly, we have our @renderMembers@ functions, which loops through the @membersList@, and append them as list items to the @ul@ tag with the id of @memberList@. 
+
+
+h2.
+  default: Step  6 – Sending and receiving calls
+
+We have been able to identify who is online and ready to receive instant messages using WebRTC data channels, now we move to the part where we send and receive calls.
+
+Before jumping into the code, it would be nice to notify users when a call is coming in, and when a call is declined. Let’s get started doing that.
+
+```[javascript]
+    function call(client_id) {
+        if (client_id === clientId) return
+        alert(`attempting to call ${client_id}`)
+        AblyRealtime.publish(`incoming-call/${client_id}`, {
+                user: clientId
+            })
+    }
+    AblyRealtime.subscribe(`incoming-call/${clientId}`, call => {
+        if (currentCall != undefined) {
+            // user is on another call
+            AblyRealtime.publish(`call-details/${call.data.user}`, {
+                user: clientId,
+                msg: 'User is on another call'
+            })
+            return
+        }
+        var isAccepted = confirm(`You have a call from ${call.data.user}, do you want to accept?`)
+        if (!isAccepted) {
+            // user rejected the call
+            AblyRealtime.publish(`call-details/${call.data.user}`, {
+                user: clientId,
+                msg: 'User declined the call'
+            })
+            return
+        }
+        currentCall = call.data.user
+        AblyRealtime.publish(`call-details/${call.data.user}`, {
+            user: clientId,
+            accepted: true
+        })
+    })
+    AblyRealtime.subscribe(`call-details/${clientId}`, call => {
+        if (call.data.accepted) {
+            initiateCall(call.data.user)
+        } else {
+            alert(call.data.msg)
+        }
+    })
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/6ddd1f92f071f89b79fda22d7d0564514e652f7d
+
+**What is going on in the code block above?**
+
+To chat with a user/member, we need to click the @call@ button next to their name. This button, in turn, calls the @call@ method, which we have just defined.
+
+First, we send an @incoming-call@ event coined by joining the @incoming-call@  string with the client’s id, to notify the other peer that someone is requesting a video call.
+
+Next, we subscribe to the @incoming-call@ event, then do a couple of checks.
+
+# check if user’s @currentCall@ is not undefined, which means he is on another call, then tell the caller the other peer is on a call.
+# verify that the user wants to pick the call. if he does not, tell the caller so, else, tell the user the call request has been accepted.
```

Comment:
'tell the caller so, else, tell the user the call request has been accepted.' -> 'tell the caller so, otherwise, tell the user that the call request has been accepted.'